pkg/api/server/v1alpha2/result/result.go

```diff
@@ -80,3 +91,23 @@ func Match(r *pb.Result, prg cel.Program) (bool, error) {
 	}
 	return b, nil
 }
+
+// GetResultByName is the helper function to get a Result by its parent and name
+func GetResultByName(gdb *gorm.DB, name string) (*pb.Result, error) {
+	parent, name, err := ParseName(name)
+	if err != nil {
+		return nil, err
+	}
+	var results []*db.Result = []*db.Result{}
+	if err := gdb.Where("parent = ?", parent).Where("name = ?", name).Find(&results).Error; err != nil {
+		log.Printf("failed to query on database: %v", err)
+		return nil, fmt.Errorf("failed to query on a result: %w", err)
+	}
+	if len(results) == 0 {
+		return nil, status.Error(codes.NotFound, "result not found")
+	}
+	if len(results) > 1 {
+		log.Println("Warning: multiple rows found")
+	}
```

Comment:
You can simplify this a bit by using the `First` operator - see https://gorm.io/docs/query.html#Retrieving-objects-with-primary-key

You might find https://github.com/tektoncd/results/pull/11/files#diff-fc57ccd1ec8e2040c7ebcf3b43f29ccdd06e4c7cda1a66bba8875c6f41902a95R61-R67 as a useful example.